lib/openhab/dsl/rules/builder.rb

```diff
@@ -964,33 +964,68 @@ def every(value, at: nil, attach: nil)
         #
         # Run this rule when the script is loaded.
         #
-        # Execute the rule on OpenHAB start up and whenever the script is
-        # reloaded. This is useful to perform initialization routines,
-        # especially when combined with other triggers.
+        # Execute the rule whenever the script is first loaded, including on OpenHAB start up,
+        # and on subsequent reloads on file modifications.
+        # This is useful to perform initialization routines, especially when combined with other triggers.
         #
-        # @param [true, false] run_on_start Run this rule on start, defaults to True
-        # @param [Object] attach object to be attached to the trigger
+        # @param [Object] attach Object to be attached to the trigger
         # @return [void]
         #
         # @example
         #   rule "startup rule" do
-        #     on_start
+        #     on_load
         #     run do
         #       <calculate some item state>
         #     end
         #   end
         #
         # @example
-        #   rule 'Ensure all security lights are on' do
-        #     on_start
+        #   rule "Ensure all security lights are on" do
+        #     on_load
         #     run { Security_Lights.on }
         #   end
         #
-        # rubocop:disable Style/OptionalBooleanParameter
-        def on_start(run_on_start = true, attach: nil)
-          @on_start = Struct.new(:enabled, :attach).new(run_on_start, attach)
+        def on_load(attach: nil)
+          raise ArgumentError, "on_load can only be called once" if @on_load # prevent overwriting the attachment
+
+          @on_load = { attachment: attach }
+        end
+
+        #
+        # Run this rule when OpenHAB reached a certain start level
+        #
+        # This will only trigger during OpenHAB start up. It won't trigger on script reloads.
```

Comment:
hmmm, this seems unusual to me. most things that deal with start level will fire immediately if that start level has already been reached.

Reply:
Based on observation, it only triggers on startup, not on reload.